modules/miner/update.go

```diff
@@ -28,11 +27,138 @@ func (m *Miner) addNewTxns(diff *modules.TransactionPoolDiff) {
 				totalFees = totalFees.Add(fee)
 			}
 		}
-		m.splitSets[m.setCounter] = &splitSet{
+		// We will check to see if this splitSet belongs in the block.
+		s := &splitSet{
 			size:         size,
 			averageFee:   totalFees.Div64(size),
 			transactions: newSet.Transactions,
 		}
+
+		elem := &mapElement{
+			set:   s,
+			id:    splitSetID(m.setCounter),
+			index: 0,
+		}
+
+		m.splitSets[splitSetID(m.setCounter)] = s
+		newElements = append(newElements, elem)
+	}
+	return newElements
+}
+
+// addMapElementTxns places the splitSet from a mapElement into the correct
+// mapHeap.
+func (m *Miner) addMapElementTxns(elem *mapElement) {
+	candidateSet := elem.set
+
+	// Check if heap for highest fee transactions has space.
+	if m.blockMapHeap.size+candidateSet.size < types.BlockSizeLimit-5e3 {
+		m.blockMapHeap.Push(elem)
+		return
+	}
+
+	// While the heap cannot fit this set s, and while the (weighted) average
+	// fee for the lowest sets from the block is less than the fee for the set
+	// s, continue removing from the heap. The block heap doesn't have enough
+	// space for this transaction. Check if removing sets from the blockMapHeap
+	// will be worth it. bottomSets will hold  the lowest fee sets from the
+	// blockMapHeap
+	bottomSets := make([]*mapElement, 0)
+	var sizeOfBottomSets uint64
+	var averageFeeOfBottomSets types.Currency
```

Comment:
So each time we add a mapElement we'll have to iterate through the transaction set and compute it's total fee and instead of using the average fee? What about changing the splitSet definition to have a `totalFee` field instead of `averageFee` like mentioned before? That way since it already stores its own size we can still compute a splitSet's average fee precisely and quickly.